# COVID-19 dataset

We have taken a small dataset of Covid-19, just for your understaning purpose. You have to work on the original dataset which contains about 19000 rows.

The data used here is till 29-April-2020 and has records as on 29-April-2020.

This data is available as a CSV file, downloaded from Kaggle.

We will analyze this data using the Pandas DataFrame.

In [4]:
# Importing libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
print('Libraries imported sucessfully!!!!')

Libraries imported sucessfully!!!!


In [5]:
# Loading the dataset
covid = pd.read_csv('Covid_dataset4.csv')
print('Loaded datset sucessfully')

Loaded datset sucessfully


In [6]:
# First five rows
covid.head()

Date State       Region  Confirmed  Deaths  Recovered
0  4/29/2020   NaN  Afghanistan       1939      60        252
1  4/29/2020   NaN      Albania        766      30        455
2  4/29/2020   NaN      Algeria       3848     444       1702
3  4/29/2020   NaN      Andorra        743      42        423
4  4/29/2020   NaN       Angola         27       2          7

In [7]:
# Shape of datset
covid.shape

(321, 6)

In [8]:
# Summary of dataset
covid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 321 entries, 0 to 320
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Date       321 non-null    object
 1   State      140 non-null    object
 2   Region     321 non-null    object
 3   Confirmed  321 non-null    int64 
 4   Deaths     321 non-null    int64 
 5   Recovered  321 non-null    int64 
dtypes: int64(3), object(3)
memory usage: 15.2+ KB


In [9]:
# Describtion of dataset
covid.describe()

Confirmed        Deaths      Recovered
count     321.000000    321.000000     321.000000
mean     9949.800623    709.152648    3030.277259
std     31923.853086   3236.162817   14364.870365
min         0.000000      0.000000       0.000000
25%       104.000000      2.000000       2.000000
50%       653.000000     12.000000      73.000000
75%      4655.000000    144.000000     587.000000
max    299691.000000  27682.000000  132929.000000

In [10]:
#Check missing value
covid.isna().sum()

Date           0
State        181
Region         0
Confirmed      0
Deaths         0
Recovered      0
dtype: int64

In the above dataset state column 181 missing entries. Too many data in state column is missing. 

## Q.1 ) Show the number of Confirmed , Deaths and Recovered cases in each Region.

In [16]:
list(covid.groupby('Region')['Confirmed','Deaths','Recovered'])

C:\Users\sai\AppData\Local\Temp\ipykernel_9340\2348070316.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  list(covid.groupby('Region')['Confirmed','Deaths','Recovered'])


[('Afghanistan',
     Confirmed  Deaths  Recovered
  0       1939      60        252),
 ('Albania',
     Confirmed  Deaths  Recovered
  1        766      30        455),
 ('Algeria',
     Confirmed  Deaths  Recovered
  2       3848     444       1702),
 ('Andorra',
     Confirmed  Deaths  Recovered
  3        743      42        423),
 ('Angola',
     Confirmed  Deaths  Recovered
  4         27       2          7),
 ('Antigua and Barbuda',
     Confirmed  Deaths  Recovered
  5         24       3         11),
 ('Argentina',
     Confirmed  Deaths  Recovered
  6       4285     214       1192),
 ('Armenia',
     Confirmed  Deaths  Recovered
  7       1932      30        900),
 ('Australia',
       Confirmed  Deaths  Recovered
  189        106       3        101
  264       3016      40       2284
  271         28       0         25
  283       1033       6        943
  297        438       4        420
  301        219      12        144
  309       1361      18       1291
  314        551

In [17]:
# In the above we have multiple entries so will take sum
covid.groupby('Region')['Confirmed','Deaths','Recovered'].sum()

C:\Users\sai\AppData\Local\Temp\ipykernel_9340\1756885763.py:2: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  covid.groupby('Region')['Confirmed','Deaths','Recovered'].sum()


Confirmed  Deaths  Recovered
Region                                          
Afghanistan              1939      60        252
Albania                   766      30        455
Algeria                  3848     444       1702
Andorra                   743      42        423
Angola                     27       2          7
...                       ...     ...        ...
West Bank and Gaza        344       2         71
Western Sahara              6       0          5
Yemen                       6       0          1
Zambia                     97       3         54
Zimbabwe                   32       4          5

[187 rows x 3 columns]

## Q2) Remove all the records where Confirmed Cases is Less Than 10.

In [19]:
(covid.Confirmed < 10).sum()

17

In [20]:
# total 17 records are present now delete this record
covid = covid[~(covid.Confirmed < 10)]

In [21]:
(covid.Confirmed < 10).sum()

0

 The records are deleted from dataset

## Q.3) In which Region, maximum number of Confirmed cases were recorded ?

In [24]:
covid.groupby('Region').Confirmed.sum().sort_values(ascending=False).head(10)

Region
US                1039909
Spain              236899
Italy              203591
France             166536
UK                 166432
Germany            161539
Turkey             117589
Russia              99399
Iran                93657
Mainland China      82861
Name: Confirmed, dtype: int64

# Q.4) In which Region, minimum number of Deaths cases were recorded ?

In [27]:
covid.groupby('Region').Deaths.sum().sort_values().head(20)

Region
Cambodia                            0
Seychelles                          0
Saint Lucia                         0
Central African Republic            0
Saint Kitts and Nevis               0
South Sudan                         0
Rwanda                              0
Grenada                             0
Macau                               0
Madagascar                          0
Nepal                               0
Namibia                             0
Saint Vincent and the Grenadines    0
Mozambique                          0
Holy See                            0
Timor-Leste                         0
Mongolia                            0
Uganda                              0
Laos                                0
Eritrea                             0
Name: Deaths, dtype: int64

## Q.5) How many Confirmed , Deaths & Recovered cases were reported from India till 29 April 2020 ?

In [28]:
covid[covid.Region == 'India']

Date State Region  Confirmed  Deaths  Recovered
74  4/29/2020   NaN  India      33062    1079       8437

## Q. 6-A ) Sort the entire data wrt No. of Confirmed cases in ascending order.

In [30]:
covid.sort_values('Confirmed')

Date      State    Region  Confirmed  Deaths  Recovered
156  4/29/2020        NaN  Suriname         10       1          8
70   4/29/2020        NaN  Holy See         10       0          2
59   4/29/2020        NaN    Gambia         10       1          8
318  4/29/2020      Yukon    Canada         11       0          0
217  4/29/2020  Greenland   Denmark         11       0         11
..         ...        ...       ...        ...     ...        ...
57   4/29/2020        NaN    France     165093   24087      48228
168  4/29/2020        NaN        UK     165221   26097          0
80   4/29/2020        NaN     Italy     203591   27682      71252
153  4/29/2020        NaN     Spain     236899   24275     132929
265  4/29/2020   New York        US     299691   23477          0

[304 rows x 6 columns]

## Q. 6-B ) Sort the entire data wrt No. of Recovered cases in descending order.

In [31]:
covid.sort_values('Recovered',ascending=False)

Date     State          Region  Confirmed  Deaths  Recovered
153  4/29/2020       NaN           Spain     236899   24275     132929
61   4/29/2020       NaN         Germany     161539    6467     120400
76   4/29/2020       NaN            Iran      93657    5957      73791
80   4/29/2020       NaN           Italy     203591   27682      71252
229  4/29/2020     Hubei  Mainland China      68128    4512      63616
..         ...       ...             ...        ...     ...        ...
258  4/29/2020    Nevada              US       4934     230          0
257  4/29/2020  Nebraska              US       3851      56          0
255  4/29/2020   Montana              US        451      16          0
254  4/29/2020  Missouri              US       7660     338          0
274  4/29/2020      Ohio              US      17303     937          0

[304 rows x 6 columns]

## 7. Find the unique countries in dataset

In [33]:
covid.Region.unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Andorra', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Austria',
       'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh', 'Barbados',
       'Belarus', 'Belgium', 'Belize', 'Benin', 'Bolivia',
       'Bosnia and Herzegovina', 'Botswana', 'Brazil', 'Brunei',
       'Bulgaria', 'Burkina Faso', 'Burma', 'Burundi', 'Cabo Verde',
       'Cambodia', 'Cameroon', 'Central African Republic', 'Chad',
       'Chile', 'Colombia', 'Congo (Brazzaville)', 'Congo (Kinshasa)',
       'Costa Rica', 'Croatia', 'Cuba', 'Cyprus', 'Czech Republic',
       'Denmark', 'Diamond Princess', 'Djibouti', 'Dominica',
       'Dominican Republic', 'Ecuador', 'Egypt', 'El Salvador',
       'Equatorial Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Fiji', 'Finland', 'France', 'Gabon', 'Gambia', 'Georgia',
       'Germany', 'Ghana', 'Greece', 'Grenada', 'Guatemala', 'Guinea',
       'Guinea-Bissau', 'Guyana', 'Haiti', 'Holy See', 'Honduras'

## 8. Total number of confirmed, death, recovered cases

In [36]:
print('The total number of confirmed cases are :',covid.Confirmed.sum())

The total number of confirmed cases are : 3193807


In [37]:
print('The total number of death cases are :',covid.Deaths.sum())

The total number of death cases are : 227633


In [41]:
print('% of death cases are :',((covid.Deaths.sum())/(covid.Confirmed.sum()))*100)

% of death cases are : 7.1273248508754605


In [42]:
print('The total recoverd cases are :',covid.Recovered.sum())

The total recoverd cases are : 831638


In [46]:
print('% of recovered cases are :',round(((covid.Recovered.sum())/(covid.Confirmed.sum()))*100,2))

% of recovered cases are : 26.04
